mscore/file.cpp

```diff
@@ -1209,9 +1208,29 @@ QString MuseScore::getFotoFilename(QString& filter, QString* selectedFilter)
       // setup side bar urls
       saveImageDialog->setSidebarUrls(urls);
 
+      // if no saves were made for current score, then use the score's name as default
```

Comment:
It looks good, but this will solve the bug when preferences.nativeDialogs is false only. That's only on Linux by default. The name needs to be computed before line 1189. I would suggest to take a look to getSaveScoreName and do exactly the same.